app/controllers/application_controller.rb

```diff
@@ -5,6 +5,142 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions unless test?
+    set :session_secret, "my_super_secret_session"
   end
 
-end
\ No newline at end of file
+  helpers do
+    def current_user(session)
+      @user = User.find(session[:id])
+      @user
+    end
+
+    def is_logged_in?(session)
```

Comment:
It's better to rely on the successful call to `current_user` than only introspecting on the `session[:id]` value. In theory, a user might have an invalid `id` in their session which would make `logged_in?` return true even though that user would not be findable later. Also, because `current_user` is memoized, it makes sense to call it as early in the request cycle as possible, which generally is the first call `logged_in?`.

```ruby
def logged_in?
  !!current_user
end
```